### ISSUE TITLE
Optimizer optimization base Adam

### ISSUE THREAD

**Issue by SevenPLF** [pytorch/pytorch#61883]
On the basis of the optimizer Adam, make changes. It can improve the optimization efficiency.

The parameter updating formula w = w + LR * G. The LR learning rate is set manually. In some new models, the setting of LR directly affects the training efficiency. If LR is too large, the training result will be divergent, and if LR is too small, the training speed will be too slow. Through each gradient update, loss is introduced. The dynamic adjustment of LR by PID adjustment method in automatic control theory can speed up the model training.

In this method, LR is evaluated by the incoming loss, so as to increase or decrease the parameters of this gradient update. It works with G.

![lr-0_001](https://user-images.githubusercontent.com/50135313/126255540-c69f6d4a-537f-4a6b-9379-63536acc379c.png)

PID regulation is a common method in the field of automation for steady-state regulation. For the disturbance of different systems, the system can achieve steady-state quickly by adjusting the proportion, integral and differential links of the system. I applied the discrete mode of PID to the regulator of training, and evaluated the learning rate by PID. The results of the regression prediction were good. But I don't have a specific paper.

Because I'm just beginning to learn the content of deep learning, I don't have many models and resources to verify. At present, it is found that it performs well in the regression task where the loss function is mselos. However, due to limited resources, I would like to put forward my own ideas here. I hope you can have a look together.

I do this to evaluate the learning rate. So it's not just Adam. For other optimizers, it should also improve efficiency.

cc @vincentqb @iramazanli

**Comment by albanD** [comment]
Closing due to inactivity.
